sklearn/metrics/scorer.py

```diff
@@ -338,10 +434,9 @@ def _check_multimetric_scoring(estimator, scoring=None):
         True if scorer is a list/tuple or dict of callables
         False if scorer is None/str/callable
     """
-    if callable(scoring) or scoring is None or isinstance(scoring,
-                                                          str):
+    if callable(scoring) or scoring is None or isinstance(scoring, str):
         scorers = {"score": check_scoring(estimator, scoring=scoring)}
-        return scorers, False
+        return _MultimetricScorer(**scorers), False
```

Comment:
Why return a `MultiMetricScorer` when there is only one scorer?

Reply:
`_check_multimetric_scoring` always returned a multimetric scorer. (On master it returned a dictionary which was the data structure used to denote "mutlimetric scoring".

Reply:
I disagree, a dict with only one key (as here) denotes a single metric scorer. That's the reason `is_multimetric` is False here.

Since no caching happens with a single-metric scorer, I think we should not change this part and still return the dict, to avoid the confusion.

Reply:
Or else, `MultiMetricScorer` should have a whole different name. It doesn't make sense to return a `MultiMetricScorer` instance while `is_multimetric` is False

Reply:
A user can pass a dictionary to `scorer` with one key and `is_multimetric` will be true.

Reply:
I do plan on removing `is_multimetric`. And have "anything that returns a dictionary" as multimetric.